iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/DirectoryCopySourceCheck.java

```diff
@@ -84,25 +83,31 @@ private enum PathSensitivity {
    * Indicate if a path is sensitive. A path is sensitive if any of the current condition is true :
    * <ul>
    * <li>if the provided path resolve to a root level or to the current directory</li>
-   * <li>if there is any top level entry (level <= 1) ending with a wildcard</li>
+   * <li>if the top level entry end with a wildcard and there is no level below</li>
    * </ul>
    * Examples of root level : '/' ; '/test/..' ; 'c:/' ; 'c:/test/..'
    * Examples of current level : '.' ; './test/..'
-   * Examples of top level entry ending with wildcard : 'a*' ; './a*' ; '/a*' ; './test/../a*'
+   * Examples of top level entry ending with wildcard and nothing behind : 'a*' ; './a*' ; '/a*' ; './test/../a*'
    */
   private static PathSensitivity isSensitivePath(String path) {
     String[] levels = normalize(path);
     if (levels.length == 0) return PathSensitivity.ROOT_OR_CURRENT;
     if (levels.length == 1 && (isRootOrCurrent(levels[0]))) return PathSensitivity.ROOT_OR_CURRENT;
-    if (Arrays.stream(levels).limit(2).anyMatch(l -> l.endsWith("*"))) return PathSensitivity.TOP_LEVEL_GLOBBING;
+    int topLevel = getTopLevelIndex(levels);
+    if (levels[topLevel].endsWith("*") && levels.length == topLevel+1) return PathSensitivity.TOP_LEVEL_GLOBBING;
     return PathSensitivity.SAFE;
   }
 
   private static boolean isRootOrCurrent(String level) {
     return (level.isEmpty() || ".".equals(level) || WINDOWS_DRIVE_PATTERN.matcher(level).find());
   }
 
-  private static String[] normalize(String path) {
+  private static int getTopLevelIndex(String[] levels) {
```

Comment:
As discussed, we can rename it to "getLevelToCheckIndex".
Also, please add a short comment about the purpose of this method